### PR TITLE
Accept empty ip to mean 'all interfaces'

### PR DESCRIPTION
**Pull request by yuvipanda**
For consistency with python/node

See discussion on https://github.com/jupyterhub/jupyterhub/pull/740

### CODE REVIEW

File: nchp/app.py [hunk 1]
```diff
@@ -185,12 +185,25 @@ def build_nginx_conf(self):
         public_ssl = self.public_ssl_cert != ''
         api_ssl = self.api_ssl_cert != ''
 
+        # '' means 'listen on all interfaces' in python / nodejs
+        # So we take it in to be compatible
+        # FIXME: check what this does for ipv6
+        if self.public_ip = '':
```
Comment (ryanlovett) [inline nchp/app.py]: s/=/==/

File: nchp/app.py [hunk 2]
```diff
@@ -185,12 +185,25 @@ def build_nginx_conf(self):
...
+            public_ip = '0.0.0.0'
+        else:
+            public_ip = self.public_ip
+
+        if self.api_ip = '':
```
Comment (ryanlovett) [inline nchp/app.py]: Here too. :)

**Comment by yuvipanda** [comment]
@ryanlovett whoops :D fixed.  I've sent you an invite for this repo, so you can merge PRs now too!

**Comment by ryanlovett** [comment]
With great power comes greater code review practices. :)

Edit: I'll let you merge this one in case you want to test it elsewhere.

**Pull request merged**